### PR TITLE
test connection UI

### PR DESCRIPTION
**Pull request by seavor**
## Related Ticket(s)
- Implements https://trello.com/c/mjcfYOF9/4-test-connection-in-add-edit-known-host-modal

Screen Shot 2022-03-19 at 3 52 28 PM

### CODE REVIEW

File: webclient/src/components/KnownHosts/KnownHosts.tsx [hunk 1]
```diff
@@ -152,20 +193,31 @@ const KnownHosts = (props) => {
     );
   };
 
+  const testConnection = () => {
+    setTestingConnection(s => ({
+      testing: true,
```
Comment (ZeldaZach) [inline webclient/src/components/KnownHosts/KnownHosts.tsx:198]: Would an ENUM be better compared to three unique booleans? Is that something TS can support?

File: webclient/src/websocket/commands/SessionCommands.ts [hunk 2]
```diff
@@ -25,6 +25,9 @@ export class SessionCommands {
       case WebSocketConnectReason.PASSWORD_RESET:
         SessionCommands.updateStatus(StatusEnum.CONNECTING, 'Connecting...');
         break;
+      case WebSocketConnectReason.TEST_CONNECTION:
+        webClient.testConnect({ ...options });
```
Comment (ZeldaZach) [inline webclient/src/websocket/commands/SessionCommands.ts:29]: Would passing options be the same as what you're doing?

File: webclient/src/components/KnownHosts/KnownHosts.tsx (hunk 1 above)
Comment (seavor) [reply webclient/src/components/KnownHosts/KnownHosts.tsx:198]: Not really, we'd then have to do value comparisons down where we use these booleans.

File: webclient/src/websocket/commands/SessionCommands.ts (hunk 2 above)
Comment (seavor) [reply webclient/src/websocket/commands/SessionCommands.ts:29]: Yes, it's just a convention to ensure immutability.

**Review by ebbit1q — APPROVED** [review]
this is working as expected, would like some input on icon choices still though

**Comment by seavor** [comment]
> this is working as expected, would like some input on icon choices still though

We'll get some feedback from Lee on a follow up.

**Pull request merged**